Fix #3456 : Full exploration title on mobile visible as info icon

kirankonduru opened the pull request:
Fixes #3456
The info icon is now visible on the footer and removes exploration titles from the top of the page for mobile devices.

kirankonduru committed:
Full exploration title on mobile visible as info icon

seanlip commented:
Thanks, @kirankonduru! Assigning to @Oishikatta for verification and review.

Oishikatta reviewed (changes requested):
This removes the info icon from the navbar entirely, but we need to remove it only if the user is on the mobile version of the site. Removing it already happens since the titles aren't being displayed now for mobile.

Angular allows using the same controller in different areas of the page, so I believe you can simply add the ng-controller with new name and the li block back to the navbar breadcrumb section in exploration_player.html. (And the .oppia-navbar-breadcrumb-icon style in oppia.css).

The other side of that is not showing the icon in the footer when not on the mobile version. One possible way that could be done is with CSS, something like this:

    @media (min-width: 768px) {
      .oppia-exploration-info-icon {
        display: none;
      }
    }

Although it would probably be best to add a distinct class name from the one in the navbar if doing that.


The info dialog uses a modal whose size is larger than some common phone screen sizes. For example testing in Chrome devtools as iPhone 5 or iPhone 6 the close button is not visible. One way to solve this would be by adding a max-width to the .modal-dialog class, e.g. 98vw.

kirankonduru committed:
Full exploration title on mobile visible as info icon fixes

kirankonduru commented:
Hi @Oishikatta 
Thanks for the review. I interpreted the comment incorrectly and thought the info icon is to be moved to the footer everywhere. 

Anyway I've made the changes suggested by you. Can you review again? Thanks

Oishikatta commented:
The PR is looking good to me, thank you!

One minor thing: Do you think we could change the Oppia logo to show the full "Oppia" instead of just "O" on mobile now that there is space for it?


There are also a couple things I'm unsure on that I'll pass to @seanlip:

Having the class "protractor-test-exploration-info-icon" on both of the info icons -- does this matter? I'm not sure if the test should be changed to .get(0), or the class removed from one, or it doesn't need a change.

CSS in directives: Should everything be prefixed where possible? I don't know if the prefixing is a code style thing or just side-effect of the initial moving from oppia.css into directives. (Referring to https://github.com/kirankonduru/oppia/blob/43d69889e95b170d8c645774f17c108553b6b959/core/templates/dev/head/pages/exploration_player/exploration_footer_directive.html#L113)

seanlip commented:
Let's use separate protractor-test classes (appropriately named) for the different icons -- I think this will reduce confusion. Alternatively, we can remove the ones that are not needed, and add them later when we implement protractor tests for mobile.

Yes, definitely please prefix. I think CSS rules used to get applied site-wide -- I'm not sure what the behaviour is now, actually, since we moved some stuff out into directives rather than jinja-including it. But it still seems like good practice to be clear about the scope of the rule and to make sure it doesn't apply outside the directive that it pertains to.

kirankonduru committed:
Full exploration title on mobile visible as info icon fixes

kirankonduru commented:
Changed the Oppia logo to show the full "Oppia" instead of just "O".
Removed protractor-test-exploration-info-icon from the footer as it is not used anywhere.
I'm sorry but I could not understand what exactly you meant by prefixing.

Oishikatta commented:
The prefixing is when using CSS in a directive, all classes should have something specific to the directive in front of them.

In exploration_footer_directive.html the prefix is ".oppia-exploration-footer", if you look at the earlier styles in that file they all start with that and then a space and the other selectors.

So for example [Line 117](https://github.com/kirankonduru/oppia/blob/ebd1abc2165bea174f89792a531e7c045e05341e/core/templates/dev/head/pages/exploration_player/exploration_footer_directive.html#L117) would change from:

`.oppia-navbar-footer-info-icon {`

To:

`.oppia-exploration-footer .oppia-navbar-footer-info-icon {`

and so on for the other new classes.

kirankonduru committed:
Prefixes css in footer

kirankonduru commented:
@Oishikatta Prefixed the css in footer. Can you review? Thanks

Oishikatta commented on core/templates/dev/head/pages/exploration_player/exploration_footer_directive.html:
```diff
@@ -114,14 +114,42 @@
+    padding: 12px;
+  }
+
+  .oppia-exploration-footer .modal-dialog {
```
It turns out that the modal-dialog is appended to the body, so this particular style can't have a prefix. Let's delete this style from here and put the max-width in the existing `.oppia-modal-information-card .modal-dialog {` block in oppia.css.

kirankonduru replied:
Done.

Oishikatta reviewed (changes requested):
Thanks, looks great! Just one tiny thing and then LGTM

kirankonduru committed:
Mobile info icon Minor css prefix fix

Oishikatta commented:
LGTM
